testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -657,7 +686,97 @@ impl SignCoordinator {
+                "threshold" => self.weight_threshold
+            );
+
+            for (message, slot_id) in messages.into_iter().zip(slot_ids) {
+                match message {
+                    SignerMessageV0::BlockResponse(BlockResponse::Accepted((
+                        response_hash,
+                        signature,
+                    ))) => {
+                        let block_sighash = block.header.signer_signature_hash();
+                        if block_sighash != response_hash {
+                            warn!(
+                                "Processed signature but didn't validate over the expected block. Returning error.";
+                                "signature" => %signature,
+                                "block_signer_signature_hash" => %block_sighash,
+                                "slot_id" => slot_id,
+                            );
+                            continue;
+                        }
+                        debug!("SignCoordinator: Received valid signature from signer"; "slot_id" => slot_id, "signature" => %signature);
+                        let Some(signer_entry) = &self.signer_entries.get(&slot_id) else {
+                            return Err(NakamotoNodeError::SignerSignatureError(
+                                "Signer entry not found".into(),
+                            ));
+                        };
+                        let Ok(signer_pubkey) =
+                            StacksPublicKey::from_slice(&signer_entry.signing_key)
+                        else {
+                            return Err(NakamotoNodeError::SignerSignatureError(
+                                "Failed to parse signer public key".into(),
+                            ));
+                        };
+                        let Ok(valid_sig) = signer_pubkey.verify(block_sighash.bits(), &signature)
+                        else {
+                            warn!("Got invalid signature from a signer. Ignoring.");
+                            continue;
+                        };
+                        if !valid_sig {
+                            warn!(
+                                "Processed signature but didn't validate over the expected block. Ignoring";
+                                "signature" => %signature,
+                                "block_signer_signature_hash" => %block_sighash,
+                                "slot_id" => slot_id,
+                            );
+                            continue;
+                        }
+                        if !gathered_signatures.contains_key(&slot_id) {
+                            total_weight_signed = total_weight_signed
+                                .checked_add(signer_entry.weight)
+                                .expect("FATAL: total weight signed exceeds u32::MAX");
+                        }
+                        debug!("SignCoordinator: Total weight signed: {total_weight_signed}");
```

Comment:
This might be helpful with a bit more info:

```rust
debug!("Added another signature to block";
             "block_signer_sighash" => %block_sighash,
             "signer_pubkey" => %signer_pubkey,
             "signer_slot_id" => slot_id,
             "signature" => %signature,
             "signer_weight" => signer_entry.weight 
             "total_weight_signed" => total_weight_signed,
             );
```

Reply:
Good call - I've added this, but I removed `signer_weight` as it otherwise goes over the "max recursion" limit.